### PR TITLE
ksvalidator - multiple files

### PR DESCRIPTION
**Pull request by VladimirSlavik**
A try at #382 - multiple files and globbing. Thus far, it's a quick hack to see if it's even possible.

Unfortunately there's something mysterious about the way warnings are handled, I can't divert them no matter what. Neither wrapping in `catch_warnings` nor replacing `showwarning` does anything.

This is only a draft, so no tests.

### CODE REVIEW

**Comment by bcl** [comment]
Do you mean the 'Found option with uppercase letters' from tests/args_names.py? Those are prints not something that gets raised. They aren't a problem as long as they have a lower case version, in which case the test will fail.

File: tools/ksvalidator.py [hunk 1]
```diff
@@ -80,42 +81,62 @@ def main(argv):
...
+                retval += 1
+                retmsg.append(_("General kickstart error in input file"))
+            except Exception as e:        # pylint: disable=broad-except
+                retval += 1
+                retmsg.append(_("General error in input file:  %s") % e)
+
+    return retval, retmsg
 
 if __name__ == "__main__":
     retval, messages = main(sys.argv[1:])
```
Comment (bcl) [inline tools/ksvalidator.py:142]: Pylint doesn't like retval here and in main, rename it to 'rc' or similar.

File: tools/ksvalidator.py [hunk 2]
```diff
@@ -80,42 +81,62 @@ def main(argv):
...
+            ksfiles.extend(glob.glob(inksfile))
+
+    # check if there are any files to check
+    if not ksfiles:
+        return (1, "No files match the patterns.")
+
+    # iterate over files to check them
+    with tempfile.TemporaryDirectory(prefix="ksvalidator-tmp-") as destdir:
+        for ksfile in ksfiles:
+            print(_("Checking kickstart file %(filename)s\n" % {"filename": ksfile}))
```
Comment (bcl) [inline tools/ksvalidator.py:102]: I am not sure why pylint is complaining about this %, we have _() strings with % in them all over the place.

Comment (bcl) [reply tools/ksvalidator.py:102]: Hm. pocketlint should be keeping that warning from showing up, maybe something odd with the GH workflow setup.

File: tools/ksvalidator.py (hunk 1 above)
Comment (VladimirSlavik) [reply tools/ksvalidator.py:142]: Done.

File: tools/ksvalidator.py (hunk 2 above)
Comment (VladimirSlavik) [reply tools/ksvalidator.py]: Doesn't show with last changes, probably a fluke of some kind.

**Comment by bcl** [comment]
Looks good to me, just needs some more tests to keep that coverage from dropping ;)

**Comment by VladimirSlavik** [comment]
I don't think I understand the coverage thing. On the site it shows everything stayed the same except for the one file which went up, but here I still see it go down.

**Comment by bcl** [comment]
> I don't think I understand the coverage thing. On the site it shows everything stayed the same except for the one file which went up, but here I still see it go down.

Yeah, that's odd. It appears to be the difference between branches covered. On the last master run it says " 2248 of 2390 branches covered (94.06%) " and on this PR it is saying " 2255 of 2398 branches covered (94.04%) " but I can't see any differences in the report so I think we can ignore that small drop :)

**Review by bcl — APPROVED** [review]
Looks good to me, let me know if you're ready to have this merged or want to wait for feedback from #382

**Comment by VladimirSlavik** [comment]
Thanks! Not yet, I think.
- The docs/manpages need updating too.
- I just realized there should be a check if the globs actually match anything.

**Comment by VladimirSlavik** [comment]
This is not dead! Rebased on master to start with a resuscitation and resolve conflict with #391.

**Comment by VladimirSlavik** [comment]
Docs are now fine, and I can't really understand the other point about no files found from globs, so I think this is fine?

**Review by bcl — APPROVED** [review]
Looks good, thanks!

**Pull request merged**